feder/cases/filters.py

```diff
@@ -30,4 +35,60 @@ def __init__(self, *args, **kwargs):
 
     class Meta:
         model = Case
-        fields = ["name", "monitoring", "institution", "created"]
+        fields = [
+            "name",
+            "monitoring",
+            "institution",
+            "created",
+            "confirmation_received",
+            "response_received",
+        ]
+
+
+class CaseReportFilter(django_filters.FilterSet):
+    monitoring = django_filters.ModelChoiceFilter(queryset=Monitoring.objects.all())
+    name = django_filters.CharFilter(
+        label=_("Institution name"),
+        field_name="institution__name",
+        lookup_expr="icontains",
+    )
+    voivodeship = VoivodeshipFilter(
+        widget=autocomplete.ModelSelect2(url="teryt:voivodeship-autocomplete")
+    )
+    county = CountyFilter(
+        widget=autocomplete.ModelSelect2(
+            url="teryt:county-autocomplete", forward=["voivodeship"]
+        )
+    )
+    community = CommunityFilter(
+        widget=autocomplete.ModelSelect2(
+            url="teryt:community-autocomplete", forward=["county"]
+        )
+    )
+    tags = django_filters.ModelMultipleChoiceFilter(
+        label=_("Tags"), field_name="tags", widget=forms.CheckboxSelectMultiple
+    )
+
+    def __init__(self, data=None, queryset=None, *, request=None, prefix=None):
+        super().__init__(data, queryset, request=request, prefix=prefix)
+        case = queryset.first()
+        self.filters["tags"].queryset = (
+            Tag.objects.filter(
+                Q(monitoring__isnull=True) | Q(monitoring=case.monitoring)
```

Comment:
Ja proponuje przenieść to do TagQuerySetManager, bo projektowaliśmy to z założeniem, że będzie się to powtarzać.

Reply:
Ok, przeniosę.